crates/bevy_ui/src/render/mod.rs

```diff
@@ -149,6 +150,7 @@ pub struct ExtractedUiNode {
     pub transform: Mat4,
     pub color: Color,
     pub rect: Rect,
+    pub uv_rect: Option<Rect>,
```

Comment:
There isn't any need for a separate `uv_rect` field here.
`prepare_uinodes` and `ExtractedUiNode` need work but I wouldn't reengineer things for this PR. Text in the UI already uses texture atlases, you should look at what it does there.